lib/update_signing_flow.py

```diff
@@ -43,3 +46,33 @@ def update_signing_flow(signflow_uri: str):
                                                        signflow_uri,
                                                        doc["uri"])
         agent_update(wrap_up_qs)
+
+def sync_signers_status(sig_flow, sh_workflow_details):
+    sh_workflow_users = sh_workflow_details["users"]
+    kaleidos_signers = get_signers(sig_flow, agent_query)
+    logger.debug(f"Syncing signers status ...")
+    for sh_workflow_user in sh_workflow_users:
+        proc_stat = sh_workflow_user["process_status"]
+        logger.debug(f"Signer {sh_workflow_user['user_email']} has process status {proc_stat}.")
+        kaleidos_signer = next(filter(lambda s: s["email"] == sh_workflow_user["user_email"], kaleidos_signers), None)
+        if kaleidos_signer["end_date"]:
+            logger.info(f"Signer {kaleidos_signer['email']} already has an end date in our db. No syncing needed.")
+        if kaleidos_signer and not kaleidos_signer["end_date"]:
```

Comment:
It's a little strange to check a dict entry and then later check if the dict is not `None`. If the worry is for possible `KeyError` exceptions, maybe we should move the `None` check to before the access?

```py
if kaleidos_signer is None:
    logger.info(f"Signer with e-mail address {kaleidos_signer['email']} not present in Kaleidos metadata: ignoring")
    continue  

if kaleidos_signer["end_date"]:
    logger.info(f"Signer {kaleidos_signer['email']} already has an end date in our db. No syncing needed.")
    continue

# the rest of the code

# or the above, but with branching `elif`s if you dislike early returns.
```